components/fxa-client/src/lib.rs

```diff
@@ -61,6 +61,7 @@ pub struct FirefoxAccount {
     client: Arc<FxAClient>,
     state: State,
     flow_store: HashMap<String, OAuthFlow>,
+    attached_clients_cache: Option<CachedResponse<Vec<http_client::GetAttachedClientResponse>>>,
```

Comment:
Not sure how it will work put in the code, but we should consider clearing this cache in response to `DeviceConnected` and `DeviceDisconnected` push notifications.

Reply:
@rfk I made changes in `push.rs`. Let me know if this is what you had in mind.